config/initializers/github_client.rb

```diff
@@ -2,12 +2,22 @@
 
 module GitHubClassroom
   def self.github_client(options = {})
-    client_options = {
-      client_id:     Rails.application.secrets.github_client_id,
-      client_secret: Rails.application.secrets.github_client_secret,
-      auto_paginate: true
-    }.merge!(options)
+    client_options = options.tap do |opts|
+      opts[:user_agent] = "GitHub Classroom"
 
+      unless options.key?(:access_token)
+        opts[:client_id]     = Rails.application.secrets.github_client_id
+        opts[:client_secret] = Rails.application.secrets.github_client_secret
+      end
```

Comment:
The old version of `github_client` would pass `client_id`, `client_secret`, and `access_token` if `access_token` was provided. Any reason for the change here?

Reply:
My understanding is that the current code-base uses client_id and client_secret to establish an access_token through the OAuth2 Auth Code Grant. If an access_token is provided, there seems to be no reason to pass in a client_id and client_secret. Currently, this happens as a side-effect of always passing in client_id and client_secret regardless.

If Classroom were using the Client Credentials grant for anything against GitHub.com or GitHub Enterprise, then it makes sense to configure those attributes. See https://github.com/octokit/octokit.rb#oauth-access-tokens and https://github.com/octokit/octokit.rb#application-authentication.

Does Classroom use the Client Credentials grant for any operations, currently?